src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingDiagnosticAnalyzer.cs

```diff
@@ -28,7 +28,7 @@ public override void Initialize(AnalysisContext context)
 
         private void AnalyzeSyntaxTree(SyntaxTreeAnalysisContext context)
         {
-            var diagnostics = RenameTrackingTaggerProvider.GetDiagnosticsAsync(context.Tree, DiagnosticDescriptor, context.CancellationToken).WaitAndGetResult(context.CancellationToken);
+            var diagnostics = RenameTrackingTaggerProvider.GetDiagnosticsAsync(context.Tree, DiagnosticDescriptor, context.CancellationToken).WaitAndGetResult_CanCallOnBackground(context.CancellationToken);
```

Comment:
why do still we need this API? shouldn't FG/BG concept no longer exposed through API since now it becomes implementation detail with JTF?

...

also, why WaitAndGetResult_CanCallOnBackground is in Workspace layer? FG/BG only makes sense editor layer and up? under editor layer, there is no UI or FG thread.

Reply:
> why do still we need this API? shouldn't FG/BG concept no longer exposed through API since now it becomes implementation detail with JTF?

It will be a goal to remove these APIs, but that's a big change on top of this pull request which is already a big change by itself. To start with, I'm going to make it so the new approach is an option and the old approach still works, and then we can remove the old approach separately.

> also, why WaitAndGetResult_CanCallOnBackground is in Workspace layer? FG/BG only makes sense editor layer and up? under editor layer, there is no UI or FG thread.

These helpers go back a long, long time.